### PR TITLE
feat: 170 use gltf exporter

### LINKED ISSUE #170: Exporting scene
### Description

As a developer using Tresjs I want to be able to download my scene and use it, in any other 3D software like blender

### Suggested solution

Creating an abstraction using the exporters provided by the addons in ThreeJs to create a global exporter.

Example: https://threejs.org/docs/index.html?q=export#examples/en/exporters/GLTFExporter

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] I agree to follow this project's [Code of Conduct](https://github.com/Tresjs/cientos/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/Tresjs/cientos/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://cientos.tresjs.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/tresjs/cientos/issues) that reports the same bug to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by JaimeTorrealba**
closes #170

### CODE REVIEW

**Review by alvarosabu — CHANGES_REQUESTED** [review]

File: src/core/misc/useGLTFExporter.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        const output = JSON.stringify( gltf, null, 2 )
+        saveString( output, `${name}.gltf` )
+      }
+    },
+    ( error: any ) => {
+      logError( 'An error happened', error )
+    },
+    options,
+  )
+  return true
```
[the remark this answers is not in the document]
Comment (JaimeTorrealba) [reply src/core/misc/useGLTFExporter.ts]: In this case not, this is a side effect function doesn't return anything. I removed the return statement

File: src/core/misc/useGLTFExporter.ts [hunk 2]
```diff
@@ -22,7 +22,7 @@ export async function useGLTFExporter(object3D: Object3D, options?: gltfExporter
 
   exporter.parse(
     objectToDownload.value,
-    ( gltf: Object3D ) => {
+    ( gltf: any ) => {
```
Comment (alvarosabu) [inline src/core/misc/useGLTFExporter.ts:25]: Sorry chamo, can't pass this with an `any`, isn't the type TresObject3D helpful here?

**Review by alvarosabu — CHANGES_REQUESTED** [review]
Sorry chamo, can't pass this with an `any`, isn't the type `TresObject3D` helpful here?

**Review by andretchen0 — COMMENTED** [review]
Nice job! Works as expected. For me, there are just a few points to look over.

**Comment by alvarosabu** [comment]
Hey @andretchen0 could you check if all the conversations where solved please?

**Comment by andretchen0** [comment]
> Hey @andretchen0 could you check if all the conversations where solved please?

Will do.

**Comment by andretchen0** [comment]
Hey @JaimeTorrealba

It looks like these are still pending:

* https://github.com/Tresjs/cientos/pull/290/files#r1415365368
* https://github.com/Tresjs/cientos/pull/290/files#r1415366462
* https://github.com/Tresjs/cientos/pull/290/files#r1415373969

😉

**Comment by JaimeTorrealba** [comment]
@andretchen0 sorry I haven't paid attention to this notification. Sorry.

Done :). And thanks, as you may guest I'm not good with the English grammar

**Comment by andretchen0** [comment]
> @andretchen0 sorry I haven't paid attention to this notification. Sorry.
> 
> Done :). And thanks, as you may guest I'm not good with the English grammar

No worries man. You're doing a lot better than me in my second language, lol.

**Review by andretchen0 — APPROVED** [review]
Looks good to me!

**Comment by alvarosabu** [comment]
Awesome, @JaimeTorrealba merge latest main to resolve conflicts and we are ready to merge this PR

**Comment by JaimeTorrealba** [comment]
@alvarosabu  done :D

File: playground/src/pages/misc/GLTFExporterDemo.vue [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import { TresCanvas, useRenderLoop } from '@tresjs/core'
+import { CameraControls, useTweakPane, useGLTFExporter } from '@tresjs/cientos'
+import { MeshStandardMaterial, TorusGeometry } from 'three'
+import { shallowRef } from 'vue'
+
+const donutMaterial = new MeshStandardMaterial({})
+const donutGeometry = new TorusGeometry(1, 0.5, 16, 32)
+const boxRef = shallowRef()
+
+const { pane } = useTweakPane()
```
Comment (andretchen0) [inline playground/src/pages/misc/GLTFExporterDemo.vue:11]: Since there's no `useTweakPane` this should probably be converted to `Leches`.

**Review by andretchen0 — CHANGES_REQUESTED** [review]
Since merging in main, the playground example is broken. It'll need to be converted to Leches.

File: playground/src/pages/misc/GLTFExporterDemo.vue (hunk 3 above)
Comment (JaimeTorrealba) [reply playground/src/pages/misc/GLTFExporterDemo.vue]: You're complete right, thanks for pointing Andre. It's already fix

**Review by andretchen0 — CHANGES_REQUESTED** [review]
Nice job. Works as advertised! Just one small typo to fix.

I'll approve the PR either way, but as an optional addition, could we also show how to download the whole Tres scene? No problem if you'd rather keep it simple.

**Review by andretchen0 — APPROVED** [review]
Great! Thanks for adding the scene example.

**Comment by JaimeTorrealba** [comment]
@alvarosabu please your approval here is necessary :)

**Comment by JaimeTorrealba** [comment]
@alvarosabu sorry for tag you again, I would like to merge this one :)

**Review by alvarosabu — APPROVED** [review]

**Pull request merged**